subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/SwiftProjectInitDescriptor.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import com.google.common.collect.Sets;
+import org.gradle.buildinit.plugins.internal.modifiers.BuildInitTestFramework;
+import org.gradle.buildinit.plugins.internal.modifiers.Language;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+
+import java.util.Set;
+
+import static org.gradle.buildinit.plugins.internal.ModuleNameBuilder.toModuleName;
+
+public abstract class SwiftProjectInitDescriptor extends LanguageLibraryProjectInitDescriptor {
+    private final TemplateOperationFactory templateOperationFactory;
+
+    public SwiftProjectInitDescriptor(TemplateOperationFactory templateOperationFactory) {
+        this.templateOperationFactory = templateOperationFactory;
+    }
+
+    @Override
+    public Language getLanguage() {
+        return Language.SWIFT;
+    }
+
+    @Override
+    public void generate(InitSettings settings, BuildScriptBuilder buildScriptBuilder, TemplateFactory templateFactory) {
+        buildScriptBuilder
+            .fileComment("This generated file contains a sample Swift project to get you started.")
+            .fileComment("For more details take a look at the Building Swift applications and libraries chapter in the Gradle");
+        configureBuildScript(settings, buildScriptBuilder);
+
+        TemplateOperation sourceTemplate = sourceTemplateOperation(settings);
+        TemplateOperation testSourceTemplate = testTemplateOperation(settings);
+
+        if (OperatingSystem.current().isLinux() || OperatingSystem.current().isWindows()) {
+            TemplateOperation testEntryPointTemplate = testEntryPointTemplateOperation(settings);
```

Comment:
I think we should consider generating the LinuxMain always since it's a sort of "best practice" and it would be nicer to do this so you can generate the project on macOS, edit the build file to add Linux and get Linux support "for free".

Reply:
I thought that too. The only problem is the code may be dead straight at the beginning. Let's generate it anyway.